### PR TITLE
Refactor code & add support for Winter.Sitemap

### PR DESCRIPTION
**Pull request by LukeTowers**
Added MultiLang capability to Winter.Sitemap for Winter.Blog / CMS Pages

### CODE REVIEW

**Comment by LukeTowers** [comment]
@mjauvin instead of having this plugin include extra support for Winter.Blog, shouldn't we make Winter.Blog support this plugin instead? It would provide a better example to people as to how to add support for their own plugins to this one.

So basically instead of having the extra MLBlog* models we'd change up the Winter.Blog models to handle that stuff

**Comment by mjauvin** [comment]
> @mjauvin instead of having this plugin include extra support for Winter.Blog, shouldn't we make Winter.Blog support this plugin instead? It would provide a better example to people as to how to add support for their own plugins to this one.
> 
> So basically instead of having the extra MLBlog* models we'd change up the Winter.Blog models to handle that stuff

Sure, I don't see why we couldn't do this.

File: Plugin.php [hunk 1]
```diff
@@ -347,39 +346,35 @@ protected function extendWinterSitemapPlugin(): void
...
+                return MLPage::resolveMenuItem($item, $url, $theme);
             }
-
             return false;
         });
 
-        Event::listen('winter.sitemap.makeUrlElement',
+        Event::listen('winter.sitemap.addItem',
             function ($definition, $xml, $pageUrl, $lastModified, $itemDefinition, $itemInfo, $itemReference, $urlElement) {
+                $defaultLocale = Locale::getDefault();
```
Comment (LukeTowers) [inline Plugin.php:358]: This should probably happen outside of the closure and be passed in via `use ($defaultLocale)`

File: classes/MLPage.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                'isActive' => rtrim($pageUrl, '/') === rtrim($url, '/'),
+                'mtime' => $page->mtime,
+            ];
+
+            $alternateLinks = [];
+            foreach ($locales as $locale => $name) {
+                if ($locale === $defaultLocale->code) {
+                    $pageUrl = $result['url'];
+                } else {
+                    $pageUrl = static::getMLPageUrl($page, $locale);
```
Comment (LukeTowers) [inline classes/MLPage.php:41]:
```suggestion
                    $pageUrl = static::getLocalizedPageUrl($page, $locale);
```

File: classes/MLPage.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    protected static function getMLPageUrl($page, $locale)
+    {
+        $translator = Translator::instance();
+
+        $page->rewriteTranslatablePageUrl($locale);
+        $url = $translator->getPathInLocale($page->url, $locale);
+
+        return (new Router)->urlFromPattern($url);
+    }
```
Comment (LukeTowers) [inline classes/MLPage.php:66]:
```suggestion
    /**
     * Gets the localized URL for the provided page
     */
    protected static function getLocalizedPageUrl(CmsPage $page, string $locale): string
    {
        $translator = Translator::instance();

        $page->rewriteTranslatablePageUrl($locale);
        $url = $translator->getPathInLocale($page->url, $locale);

        return (new Router)->urlFromPattern($url);
    }
```

**Pull request merged**